Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs

```diff
@@ -288,12 +288,12 @@ public override void HandleNetworkMessage(ComponentMessage message, INetChannel
                         var entity = _entityManager.GetEntity(remove.EntityUid);
                         if (entity != null && storage.Contains(entity))
                         {
-                            Remove(entity);
 
                             var item = entity.GetComponent<ItemComponent>();
                             if (item != null && playerentity.TryGetComponent(out HandsComponent hands))
                             {
-                                if (hands.PutInHand(item))
+                                if (hands.CanPutInHand(item) && hands.PutInHand(item))
+                                    Remove(entity);
                                     return;
```

Comment:
Either this belongs in the above block, in which case you should wrap both statements in braces `{ }`, or you should de-indent it.